File: p005/p5.jl
#!/usr/bin/env julia
# Project Euler Problem #5
# find smallest number that can be evenly divided by 1-20

function test_factor(n)
  for i in 11:20
    if n % i !=0
      return false
    end
  end  
  return true
end

i = 0
result = 0
while result == 0
    i+=20
    if test_factor(i)
        result = i
    end
end
    println ("The smallest number evenly divided by 1-20 is: $(result)")

File: p005/p5.lua
#!/usr/bin/env lua

--[[
Project Euler Problem #5
find smallest number that can be evenly divided by 1-20
]]


function testFactor(n)
    local num = n
    local result = true
    for i = 11, 19, 1 do
      if num % i ~= 0 then
        result = false
        break
      end
    end
    return result
end

-- Main Program --

i = 0
while true do
    i = i+20
    if testFactor(i) then
        break
    end
end

print ("The smallest number evenly divided by 1-20 is: " .. i)

